i18n tooling and translation

OmarIthawi opened the pull request:
We'd like to add Japanese as a supported language, also supported JavaScript translations via 
[`django-statici18n`](https://github.com/zyegfryed/django-statici18n/pull/41).

Before you continue, take a look at [edx/i18n-tools](https://github.com/edx/i18n-tools) and [edx/cookiecutter-django-app](https://github.com/edx/cookiecutter-django-app).

### Summary of the Changes
 - Added a Makefile to help manage the translations. The file is copied and modified from: [edx/cookiecutter-django-app](https://github.com/edx/cookiecutter-django-app)
 - Created a `locale` link to `google_drive/translations` to allow the `i18n_tools` find the po/mo files
 - Added `google_drive/translations/config.yaml` which works as the configuration file of the `i18n_tools` command 
 - Added `requirements-dev.txt` to pin development packages. Used for now only for translations.

### TODO
 - [x] Support both `django`
    * `djangojs` translation domains is ignored because the only string there's is for testing purposes
 - [x] Make Travis happy
 - [x] Document how to update the strings properly from Transifex
 - [x] Manually test that the dummy Esperanto translation is working
 - [x] Support [adding locale to the embed code](http://henrythiele.blogspot.com/2011/11/embedding-google-calendar-in-another.html) via `hl=ja_JP` parameter e.g.
   ```
   http://www.google.com/calendar/embed?src=jkloc408t9h5rdv686c8d567hs%40group.calendar.google.com&ctz=America/Chicago&hl=es
   ```
 - [x] Add tests for the calendar localization :point_up: 
 

### Screenshot
<img src="https://user-images.githubusercontent.com/645156/46402508-951a6180-c708-11e8-9a75-393dcf79ad42.png" width="450">

edx-webhook commented:
Thanks for the pull request, @OmarIthawi! I've created [OSPR-2592](https://openedx.atlassian.net/browse/OSPR-2592) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

OmarIthawi commented:
@mduboseedx Could you please let me know why the `open-source-contribution` label have been removed?

mduboseedx commented:
@OmarIthawi I actually deleted those labels but that may be due to my own confusion with the edx-solutions repo. I can easily add those back though

OmarIthawi commented:
Thanks @mduboseedx. I'll tag the latest author on this repo.

OmarIthawi commented:
@jmbowman Could you please take a look? or do you know someone else who could review this PR?

OmarIthawi commented:
Just a gentle reminder @jmbowman.

OmarIthawi commented on Makefile:
```diff
@@ -1,4 +1,4 @@
-.PHONY: clean coverage docs help quality requirements selfcheck test test-all upgrade validate
+.PHONY: clean coverage docs help quality requirements selfcheck test test-all upgrade validate dummy_translations extract_translations pull_translations push_translations
```
Done! I've also sorted them alphabetically.

OmarIthawi commented on google_drive/google_calendar.py:
```diff
@@ -18,34 +18,42 @@
 
+def _(text):
+    """
+    Dummy ugettext.
```
Well, because XBlocks are initialized as soon as possible and that makes Django freaks out with an exception!

All the other XBlocks uses this trick to overcome this limitation. Honestly, this a pretty useless trick, because XBlocks don't yet support i18n for field name, so :man_shrugging: 

But everyone's doing this in the hope that we can do something about it soon, I'm also optimistic about that as well so I do the same.

jmbowman replied:
Hmm.  Would `ugettext_lazy` work? https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#lazy-translation  If that breaks something in the depths of the XBlock machinery, I'm fine with this as is.

OmarIthawi commented on requirements/dev.txt:
```diff
@@ -4,95 +4,106 @@
 #
--e git+https://github.com/edx/xblock-sdk.git@v0.1.4#egg=xblock-sdk==0.1.4
--e git+https://github.com/edx/xblock-utils.git@v1.0.5#egg=xblock-utils==1.0.5
+-e git+git://github.com/edx/xblock-sdk.git@v0.1.2#egg=xblock-sdk==v0.1.2
```
Probably unintentionally, I'm updating this to upgrade Django

OmarIthawi commented on requirements/dev.txt:
```diff
@@ -4,95 +4,106 @@
+distlib==0.2.7            # via caniusepython3
 django-pyfs==2.0          # via xblock
-django==1.11.10
+django==1.8.19
```
I think so, reverting back

OmarIthawi commented:
Thanks @jmbowman for the review! I've addressed your comment and added a couple of commits.

Could you please take a look? Let me know if you think everything is good so I can squash the tests.

OmarIthawi commented:
Thanks @jmbowman! I've squashed the changes. Please feel free to merge.

OmarIthawi commented:
Thank you again @jmbowman!
